src/Core/src/Platform/Android/EntryExtensions.cs

```diff
@@ -47,27 +48,52 @@ public static void UpdateIsPassword(this AppCompatEditText editText, IEntry entr
 
 		internal static void SetInputType(this AppCompatEditText editText, IEntry entry)
 		{
-			editText.InputType = InputTypes.ClassText;
-			editText.InputType |= InputTypes.TextFlagMultiLine;
+			if (entry.IsReadOnly)
+				editText.InputType = InputTypes.Null;
+			else
+			{
+				var keyboard = entry.Keyboard;
+				var nativeInputTypeToUpdate = keyboard.ToInputType();
 
-			if (entry.IsPassword && ((editText.InputType & InputTypes.ClassText) == InputTypes.ClassText))
-				editText.InputType |= InputTypes.TextVariationPassword;
+				if (!(keyboard is CustomKeyboard))
```

Comment:
Why is this only when _not_ a custom keyboard?Do we expect the keyboard to override this property?